src/components/Home.js

```diff
@@ -1,7 +1,69 @@
-const Home = () => (
-  <div>
-    <h1>Home</h1>
-  </div>
-);
+import { useEffect } from 'react';
+import { Link } from 'react-router-dom';
+import { useSelector, useDispatch } from 'react-redux';
+import { fetchData, getId, setSearch } from '../redux/home/homeSlice';
+import '../styles/Home.css';
+import enter from '../images/arrow-right.png';
+
+const Home = () => {
+  const {
+    isPending, error, data,
+  } = useSelector((state) => state.home);
+  const { search } = useSelector((state) => state.home);
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    dispatch(fetchData());
+    dispatch(setSearch(''));
+  }, [dispatch]);
+
+  const handleGetId = (id) => {
+    dispatch(getId(id));
+  };
+
+  const handleSearch = (e) => {
+    dispatch(setSearch(e.target.value));
+  };
```

Comment:
- [ ] Please make sure every page (the main page and the pages for each item) should have a unique route as requested in the project requirements. Currently, All stock's pages are showing on the same route`(/details)`, For a unique route you can use the names of stocks and pass them in the URL to use in the route. You can read more about React dynamic routing [here](https://dev.to/shubhamtiwari909/dynamic-routes-in-react-491g)

![image](https://user-images.githubusercontent.com/83803414/225131891-6660557e-5184-435e-8410-7aba3b72c4ae.png)

![image](https://user-images.githubusercontent.com/83803414/235317909-d596596e-3733-4c6a-9a5a-75927d6c2eec.png)

![image](https://user-images.githubusercontent.com/83803414/235317917-210fd4fc-96a7-41cb-9893-41d4a3290df5.png)